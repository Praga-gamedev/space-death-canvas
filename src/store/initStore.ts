import { resetContext, getContext } from 'kea';
import thunkPlugin from 'kea-thunk';
import localStoragePlugin from 'kea-localstorage';

import { Reducer } from 'redux';
import reduxReset from 'redux-reset';

import { connectRouter, routerMiddleware } from 'connected-react-router';

import { createBrowserHistory } from 'history';

export const history = createBrowserHistory();

resetContext({
    createStore: {
        reducers: {
            router: connectRouter(history) as Reducer,
        },

        middleware: [routerMiddleware(history)],
        enhancers: [reduxReset()],
    },
    plugins: [
        thunkPlugin,
        localStoragePlugin({
            prefix: 'app',
        }),
    ],
});

export default getContext().store;
