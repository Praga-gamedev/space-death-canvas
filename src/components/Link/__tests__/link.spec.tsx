import React from 'react';
import { mount } from 'enzyme';
import { Link } from '../index';

import { history } from '@store/initStore';
import renderer from 'react-test-renderer';

const path = '/test';

// Пытался замокать createBrowserHistory, который изначально создается в сторе, но не получилось
// Мокаю, чтобы по клику на ссылку понять - записалась ли история или нет
jest.mock('@store/initStore', () => ({
    history: {
        push: jest.fn(),
    },
}));

describe('Link component', () => {
    test('link rendered correctly', () => {
        const children = 'Snapshot!';
        const LinkComponent = renderer
            .create(<Link path={path}>{children}</Link>)
            .toJSON();
        expect(LinkComponent).toMatchSnapshot();
    });

    test('link content rendered correctly', () => {
        const linkText = 'Click me';
        const LinkComponent = mount(
            <Link path={path}>
                <span>{linkText}</span>
            </Link>
        );

        expect(LinkComponent.find('span')).toHaveLength(1);
        expect(LinkComponent.find('span').at(0).text()).toBe(linkText);
    });

    test('change browser history onClick', () => {
        const LinkComponent = mount(<Link path={path} />);
        LinkComponent.find('label').simulate('click');

        expect(history.push).toBeCalledTimes(1);
    });
});
