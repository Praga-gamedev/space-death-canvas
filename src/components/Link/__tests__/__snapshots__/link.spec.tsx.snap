// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link component link rendered correctly 1`] = `
<label
  className="css-1gpyte2"
  onClick={[Function]}
>
  Snapshot!
</label>
`;
